test/forked/BaseForkedTest.sol

```diff
@@ -71,23 +42,31 @@ contract SmartWallet {
     }
 }
 
-contract BaseTest is Test {
+contract BaseForkedTest is Test {
+    // Nothing special about this block, it's the latest at the time of writing.
+    uint256 constant MAINNET_FORKED_BLOCK = 22947477;
```

Comment:
why hardcoding a block? wouldn't the test work on the latest always? By adding a block you make the requirements of the type of node a bit more demanding (I now know why you called archive to the env when I've been using normal nodes with this project).

Reply:
okay, I will approve, in general probably all test should be local anyways, and could have mocked any dependency like the ENS (or better, removed if we don't care about it)